src/Mod/Start/App/DisplayedFilesModel.cpp

```diff
@@ -190,16 +190,50 @@ void DisplayedFilesModel::addFile(const QString& filePath)
     if (!qfi.isReadable()) {
         return;
     }
+
+    // Check if FreeCAD can open the file by its extension
     if (!freecadCanOpen(qfi.suffix())) {
         return;
     }
+
+    // Add file information to cache
     _fileInfoCache.emplace_back(getFileInfo(filePath.toStdString()));
+
+    // If the file is a FreeCAD file (.FCStd)
     if (qfi.completeSuffix() == QLatin1String("FCStd")) {
+        // Load thumbnail from FreeCAD file
         auto thumbnail = loadFCStdThumbnail(filePath.toStdString());
         if (!thumbnail.isEmpty()) {
             _imageCache.insert(filePath, thumbnail);
         }
     }
+    else {
+        // If it is not a FreeCAD file, generate thumbnail using F3D
+        QString thumbnailPath = QString(QLatin1String("%1/%2_thumbnail.png"))
+                                    .arg(QLatin1String("/tmp"))
+                                    .arg(qfi.baseName());  // Temporary path for the thumbnail
+
+        QString command =
+            QString(QLatin1String("f3d --config=thumbnail --load-plugins=occt --verbose=quiet "

```

Comment:
this seems like it just assumes this is installed and in PATH, do we need it at all? isn't freecad itself capable of generating these thumbnails? Also is this happening in the background? because otherwise it could be quite the slowdown to generate these thumbnails.

Reply:
I know it currently doesn't do it, my point was that it should be possible to make it although I understand if you are not interested in implementing that. My other points still stand though

Reply:
if f3d will be used it's path should be configurable because on mac and windows it's not as common to have 3rd party executables in your path.